lib/rouster/deltas.rb

```diff
@@ -327,7 +327,7 @@ def get_packages(cache=true, deep=true)
       end
 
     elsif os.eql?(:rhel)
-      raw = self.run('rpm -qa --qf "%{n}@%{v}@%{arch}\n"')
+      raw = self.run('rpm -qa --qf "%{n}@%{v}@%{arch}\n"', 0, false)
```

Comment:
When sudo is enabled, the final command is modified like this:
```
:final_command => sudo ? sprintf( 'sudo bash -c "%s"', command ) : command
```
Which is a problem for commands that already contain double quotes. In most places I fixed by removing the unnecessary quotes. On this line, I fixed by not having it use sudo. I presume `dpkg-query` & `pkgutil` would be fine since they don't use double quotes in their invocation.

Reply:
i know we talked briefly about using `%Q`, but i can't recall why we didn't go in that direction..

something like
```rb
  :final_command => sudo ? sprintf(%Q(sudo bash -c '%s', command) : command
```

or maybe

```rb
  :final_command => sudo ? %Q(sudo bash -c '#{command}') : command
```

Reply:
That is also a valid solution, but doesn't fully resolve the issue. It just makes sudo mutually exclusive with _single_ quotes instead of _double_ quotes.